receiver/couchdbreceiver/client_test.go

```diff
@@ -0,0 +1,291 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package couchdbreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/couchdbreceiver"
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/config/configtls"
+	"go.uber.org/zap"
+)
+
+func defaultConfig(t *testing.T, endpoint string) client {
```

Comment:
This is easily confused w/ [createDefaultConfig](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/6ac3ae0abd49f255bcac83ccb2ccaa09cc753a29/receiver/couchdbreceiver/factory.go#L43), and doesn't actually create a config. Rather, it creates a client. Maybe call it `defaultClient`, and instantiate the client with the config provided by `createDefaultConfig` (rather than hardcode the default config again here).

Reply:
I renamed `defaultConfig` -> `defaultClient` and instantiated createDefaultClient within it.